src/python/cmd.py

```diff
@@ -17,22 +18,30 @@ def translate(src, **flags):
     if flags.get("parse"):
         print(ast)
         return ast
-    iir = IRNoder().make(ast)
+    noder = IRNoder()
+    block = noder.block
+    iir = noder.make(ast)
     if flags.get("make_ir"):
         print(IRPrinter().to_string(iir))
         return iir
     tc = TypeChecker()
     tc.check(iir)
     if flags.get("typecheck"):
         print(IRPrinter().to_string(iir))
+        if flags.get("cf_diagram"):
```

Comment:
this shouldn't be inside the `--typecheck` flag. `--make-ir` also produces IRs that may warrant a cf diagram